4.sc_Morphology_Neural_Network_MLP_Model/0.hyperparameter_optimization/scripts/Hyperparameter_Optimization_model_regression.py

```diff
@@ -0,0 +1,341 @@
+
+# %% [markdown]
+# ##### Regression Model Data Wrangling and Set Up
+# comment out if not using regression
+
+# %%
+# Parameters
+CELL_TYPE = "PBMC"
+CONTROL_NAME = "DMSO_0.100_DMSO_0.025"
+# TREATMENT_NAME = "LPS_100.000_DMSO_0.025"
+TREATMENT_NAME = "LPS_100.000_DMSO_0.025"
+MODEL_NAME = "DMSO_0.025_vs_LPS_100"
+
+# %%
+ml_configs_file = pathlib.Path("../../MLP_utils/regression_config.toml").resolve(
+    strict=True
+)
+ml_configs = toml.load(ml_configs_file)
+params = Parameters()
+mlp_params = parameter_set(params, ml_configs)
+
+# overwrite params via command line arguments from papermill
+mlp_params.CELL_TYPE = CELL_TYPE
+mlp_params.MODEL_NAME = MODEL_NAME
+mlp_params.CONTROL_NAME = CONTROL_NAME
+mlp_params.TREATMENT_NAME = TREATMENT_NAME
+mlp_params.SHUFFLE = False
+
+# %%
+# Import Data
+# set data file path under pathlib path for multi-system use
+
+# Commented out for now, using a different data set to trobleshoot
+# file_path = pathlib.Path(
+#     f"../../../data/{mlp_params.CELL_TYPE}_preprocessed_sc_norm.parquet"
+# ).resolve(strict=True)
+
+file_path = pathlib.Path(
+    "../../../data/PBMC_subset_sc_norm_DMSO_0.100_DMSO_0.025_LPS_100.000_DMSO_0.025.parquet"
+)
+
+# set path for nomic data
+nomic_df_path = pathlib.Path(
+    f"../../../2.Nomic_nELISA_Analysis/Data/clean/Plate2/nELISA_plate_430420_{mlp_params.CELL_TYPE}_cleanup4correlation.csv"
+).resolve(strict=True)
+
+df = pq.read_table(file_path).to_pandas()
+nomic_df = pd.read_csv(nomic_df_path)
+
+# %%
+# change the nomic df to standard scaler
+# select the columns that contain "NSU"
+nomic_df_scaled = nomic_df.filter(regex="NSU")
+# standardize the nomic data
+# scaler = preprocessing.StandardScaler()
+# nomic_df_scaled = pd.DataFrame(scaler.fit_transform(nomic_df_scaled), columns=nomic_df_scaled.columns)
+# add the nomic data metadata back
+nomic_df_scaled[
+    [
+        "Metadata_position_x",
+        "oneb_Metadata_Treatment_Dose_Inhibitor_Dose",
+        "fourb_Metadata_Treatment_Dose_Inhibitor_Dose",
+    ]
+] = nomic_df[
+    [
+        "Metadata_position_x",
+        "oneb_Metadata_Treatment_Dose_Inhibitor_Dose",
+        "fourb_Metadata_Treatment_Dose_Inhibitor_Dose",
+    ]
+]
+nomic_df = nomic_df_scaled.copy()
+del nomic_df_scaled
+
+# %%
+print(df.shape)
+df = pd.merge(
+    df,
+    nomic_df,
+    left_on=[
+        "Metadata_Well",
+        "oneb_Metadata_Treatment_Dose_Inhibitor_Dose",
+        "fourb_Metadata_Treatment_Dose_Inhibitor_Dose",
+    ],
+    right_on=[
+        "Metadata_position_x",
+        "oneb_Metadata_Treatment_Dose_Inhibitor_Dose",
+        "fourb_Metadata_Treatment_Dose_Inhibitor_Dose",
+    ],
+).drop(["Metadata_position_x"], axis=1)
+print(nomic_df.shape)
+print(df.shape)
+
+# %%
+# Code snippet for metadata extraction by Jenna Tomkinson
+df_metadata = list(df.columns[df.columns.str.contains("Metadata")])
+
+# define which columns are data and which are descriptive
+df_descriptive = df[df_metadata]
```

Comment:
Consider passing the value of `df_metadata` as a direct reference to `df_descriptive` instead of creating a variable which will only be used once.

Reply:
good point